spatial-tracking-explainer.md

```diff
@@ -208,35 +208,31 @@ An `XRRay` object includes both an `origin` and `direction`, both given as `DOMP
 ### Application-supplied transforms
 Frequently developers will want to provide an additional, artificial transform on top of the user's tracked motion to allow the user to navigate larger virtual scenes than their tracking systems or physical space allows. This effect is traditionally accomplished by mathematically combining the API-provided transform with the desired additional application transforms. WebXR offers developers a simplification to ensure that all tracked values, such as viewer and input poses, are transformed consistently.
 
-Developers can specify application-specific transforms by setting the `originOffset` attribute of any `XRReferenceSpace`. The `originOffset` is initialized to an identity transform, and any values queried using the `XRReferenceSpace` will be offset by the `position` and `orientation` the `originOffset` describes. The `XRReferenceSpace`'s `originOffset` can be updated at any time and will immediately take effect, meaning that any subsequent poses queried with the `XRReferenceSpace` will take into account the new `originOffset`. Previously queried values will not be altered. Changing the `originOffset` between pose queries in a single frame is not advised, since it will cause inconsistencies in the tracking data and rendered output.
+Developers can specify application-specific transforms by calling the `getOffsetReferenceSpace()` method of any `XRReferenceSpace`. This returns a new `XRReferenceSpace` with an "effective origin" equal to the native origin of the parent reference space offset by the by the `position` and `orientation` described by the `XRRigidTransform` passed to `getOffsetReferenceSpace()` plus any offset applied to the reference space previously if it was also created with `getOffsetReferenceSpace()`.
 
 A common use case for this attribute would be for a "teleportation" mechanic, where the user "jumps" to a new point in the virtual scene, after which the selected point is treated as the new virtual origin which all tracked motion is relative to.
 
 ```js
 // Teleport the user a certain number of meters along the X, Y, and Z axes
 function teleport(deltaX, deltaY, deltaZ) {
-  let currentOrigin = xrReferenceSpace.originOffset;
-  xrReferenceSpace.originOffset = new XRRigidTransform(
-      { x: currentOrigin.position.x + deltaX,
-        y: currentOrigin.position.y + deltaY,
-        z: currentOrigin.position.z + deltaZ },
-      currentOrigin.orientation);
+  xrReferenceSpace = xrReferenceSpace.getOffsetReferenceSpace(
```

Comment:
I'd love to see a super-specific example of how a reference space and an offset reference space interact, not just how they can be used.  Would it make sense to add something like this?

For example:
offsetSpace = xrReferenceSpace.getOffsetReferenceSpace(new XRRigidTransform({x: 1, y:0, z:0});
// (0,0,0) in offsetSpace coordinates is (1,0,0) in xrReferenceSpace coordinates.

Similar for chained offsets to make it extra clear how they are chained.
offsetSpace1 = xrReferenceSpace.getOffsetReferenceSpace(new XRRigidTransform({x: 1,y: 0, z: 0});
offsetSpace2 = offsetSpace1.getOffsetReferenceSpace(<rotation around y axes by 180>);
(0,0,0) in offsetSpace2 is (0,0,0) in offsetspace1 is (1, 0, 0) in xrReferenceSpace, but
(1,0,0) in offsetSpace2 is (-1,0,0) in offsetSpace1 is (0,0,0) in xrReferenceSpace.